pkg/cmd/fixcerts/fixcerts.go

```diff
@@ -0,0 +1,407 @@
+		Timeout:               5 * time.Minute,
+	}
+
+	cmd := &cobra.Command{
+		Use: "fix-certs",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Complete()
+			if err != nil {
+				return err
+			}
+
+			err = o.Validate()
+			if err != nil {
+				return err
+			}
+
+			err = o.Run()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	// TODO: add/prettify descriptions
+	cmd.Flags().StringVar(&o.KubeApiserverImage, "kube-apiserver-image", o.KubeApiserverImage, "")
+	cmd.Flags().StringVar(&o.StaticPodResourcesDir, "static-pod-resources", o.StaticPodResourcesDir, "path to store a directory containing static pod resources for recovery apiserver manifest")
+	cmd.Flags().StringVar(&o.PodManifestDir, "pod-manifest-dir", o.PodManifestDir, "directory for the static pod manifes")
+	cmd.Flags().DurationVar(&o.Timeout, "timeout", o.Timeout, "timeout, 0 means infinite")
+
+	return cmd
+}
+
+func (o *Options) Complete() error {
+	return nil
+}
+
+func (o *Options) Validate() error {
+	if o.KubeApiserverImage == "" {
+		return errors.New("missing required flag --kube-apiserver-image")
+	}
+
+	if o.PodManifestDir == "" {
+		return errors.New("pod-manifest-dir shall not be empty")
+	}
+
+	return nil
+}
+
+func (o *Options) Run() error {
+	ctx, cancel := watch.ContextWithOptionalTimeout(context.Background(), o.Timeout)
+	defer cancel()
+	// TODO: hook up signals
+
+	apiserver := &recovery.Apiserver{
+		KubeApiserverImage:    o.KubeApiserverImage,
+		PodManifestDir:        o.PodManifestDir,
+		StaticPodResourcesDir: o.StaticPodResourcesDir,
+	}
+
+	err := apiserver.Create()
+	if err != nil {
+		return fmt.Errorf("failed to create recovery apiserver: %v", err)
+	}
+	defer apiserver.Destroy()
+
+	klog.Info("Waiting for recovery apiserver to come up")
+	err = apiserver.WaitForHealthz(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to wait for recovery apiserver to be ready: %v", err)
+	}
+	klog.Info("Recovery apiserver is up")
+
+	// Run cert recovery
+
+	// We already have kubeClient created
+	kubeClient, err := apiserver.GetKubeClientset()
+	if err != nil {
+		return fmt.Errorf("failed to get kubernetes clientset: %v", err)
+	}
+
+	restConfig, err := apiserver.RestConfig()
+	if err != nil {
+		return fmt.Errorf("failed to get kubernetes rest config: %v", err)
+	}
+
+	operatorConfigClient, err := operatorversionedclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	configClient, err := configeversionedclient.NewForConfig(restConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create config client: %v", err)
+	}
+
+	certRotationCtx, certRotationCtxCancel := context.WithCancel(ctx)
+	defer certRotationCtxCancel()
+
+	certRotationWg := sync.WaitGroup{}
+
+	operatorConfigInformers := operatorexternalinformers.NewSharedInformerFactory(operatorConfigClient, 10*time.Minute)
+	certRotationWg.Add(1)
+	go func() {
```

Comment:
This is an odd structure.  `Start` is non-blocking.  I don't see a reason to wrap it in a gofunc